paasta_tools/paasta_cli/cmds/local_run.py

```diff
@@ -574,40 +619,42 @@ def configure_and_run_docker_container(docker_client, docker_hash, service, args
     )
 
 
-def validate_environment():
-    """Validates whether the current directory is good for running
-    paasta local_run"""
-    if os.getcwd() == os.path.expanduser("~"):
-        sys.stderr.write(
-            'ERROR: Don\'t run this command from your home directory.\n'
-            'Try changing to the root of your working copy of the service.\n'
-        )
-        sys.exit(1)
-    if not os.path.isfile(os.path.join(os.getcwd(), 'Dockerfile')):
-        sys.stderr.write(
-            'ERROR: No Dockerfile in the current directory.\n'
-            'Are you in the root folder of the service directory? Does a Dockerfile exist?\n'
-        )
-        sys.exit(1)
+def local_makefile_present():
+    return True
 
 
 def paasta_local_run(args):
-    validate_environment()
+    # Try to autodetect if we should build the image locally or not
+    if local_makefile_present() and args.build is None and args.pull is None:
+        sys.stderr.write("Local Makefile detected, building a local Docker image. Use --pull to override.\n")
+        build = True
+    elif args.pull is not None:
+        build = False
+    else:
+        build = True
 
     service = figure_out_service_name(args, soa_dir=args.yelpsoa_config_root)
-
     base_docker_url = get_docker_host()
-
     docker_client = Client(base_url=base_docker_url)
 
-    default_tag = 'paasta-local-run-%s-%s' % (service, get_username())
-    tag = os.environ.get('DOCKER_TAG', default_tag)
-    os.environ['DOCKER_TAG'] = tag
-
-    paasta_cook_image(None, service=service, soa_dir=args.yelpsoa_config_root)
+    if build:
+        default_tag = 'paasta-local-run-%s-%s' % (service, get_username())
```

Comment:
Should we enhance get_username to look for an environment variable like SUDO_USER so it's easier to find out who to bug if a container starts to do evil things to a machine?